lib/core.mli

```diff
@@ -536,56 +533,35 @@ val use_state : (unit -> 'state) -> 'state * (('state -> 'state) -> unit)
     let use_state initial = use_state_internal Imports.react initial]
 
 val use_reducer :
-  ('state -> 'action -> 'state) -> 'state -> 'state * ('action -> unit)
-  [@@js.custom
-    let (use_reducer_internal :
-             Imports.react
-          -> ('state -> 'action -> 'state)
-          -> 'state
-          -> 'state * ('action -> unit) ) =
-     fun (react : Imports.react) (reducer : 'state -> 'action -> 'state)
-         (init : 'state) ->
-      let result =
-        Ojs.call
-          (Imports.react_to_js react)
-          "useReducer"
-          [| Ojs.fun_to_js 2 (fun (state : Ojs.t) (action : Ojs.t) ->
-                 Obj.magic (reducer (Obj.magic state) (Obj.magic action)) )
-           ; Obj.magic init |]
-      in
-      (Obj.magic (Ojs.array_get result 0), Obj.magic (Ojs.array_get result 1))
-
-    let use_reducer reducer initial =
-      use_reducer_internal Imports.react reducer initial]
-
-val use_reducer_with_map_state :
```

Comment:
What happened with this function? It is useful to do lazy initialization https://reactjs.org/docs/hooks-reference.html#lazy-initialization

Reply:
`use_reducer` does only lazy initialization instead now. Do you think there's a reason it should need to do eager initialization?

Reply:
> Do you think there's a reason it should need to do eager initialization?

I can't think of any reasons.

> This is probably the change I'm most uncertain about, from an idealist perspective at least.

It is funny because this is one of the less disruptive changes imo 😄 I believe we have different use cases in mind.

Reply:
It's not that I have a use case in mind, more of a concern that there _ought to be_ a use case for this very weird React API. I just can't understand how that API came about, other than there being some practical reason for it.